NetflixClone/Classes/Networking/DiscoverMoviesRemote.swift

```diff
@@ -8,14 +8,57 @@
 import Foundation
 
 class DiscoverMoviesRemote {
-    typealias config = NetworkConfiguration
-    
+    typealias Config = NetworkConfiguration
     let networking: NetworkingManager = .shared
-    func discoverMovies(at page: Int) async throws -> [Movie] {
-        let url = URL(string:"\(config.baseURL)3/discover/movie?api_key=\(config.apiKey)&page=\(page)")!
-        
+    // MARK: - Discover movies (/discover/movie)
+    // https://developers.themoviedb.org/3/discover/movie-discover
+    func discoverMovies(at page: Int ) async throws -> [Movie] {
+        let url = URL(string: "\(Config.baseURL)discover/movie?api_key=\(Config.apiKey)&page=\(page)")!
         let urlRequest = URLRequest(url: url)
-        let responseModel = try await networking.responseData(urlRequest) as PaginatedMovie
+        let responseModel = try await networking.responseData(urlRequest) as PaginatedList<Movie>
+        return responseModel.results
+    }
+    enum MediaType: String {
+        case all
+        case movie
+        case tv
+        case person
+    }
+    enum TimeWindow: String {
+        case day
+        case week
+    }
+    func trendingMovies(at page: Int, type: MediaType, time: TimeWindow ) async throws -> [Movie] {
+        let urlString = "\(Config.baseURL)trending/\(type.rawValue)/\(time.rawValue)?api_key=\(Config.apiKey)&page=\(page)"
```

Comment:
We can use URL components to construct the URL instead of the repetitive URLs 
More: https://www.swiftbysundell.com/articles/constructing-urls-in-swift/